extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/CDIDelegatingTransactionManager.java

```diff
@@ -0,0 +1,176 @@
+package io.quarkus.narayana.jta.runtime;
+
+import java.io.*;
+
+import javax.enterprise.context.*;
+import javax.enterprise.event.*;
+import javax.enterprise.inject.*;
+import javax.inject.*;
+import javax.transaction.*;
+
+/**
+ * A delegating transaction manager which receives an instance of Narayana transaction manager
+ * and delegates all calls to it.
+ * On top of it the implementation adds the CDI events processing for {@link TransactionScoped}.
+ */
+@Singleton
+public class CDIDelegatingTransactionManager implements TransactionManager, Serializable {
```

Comment:
I think you might need to register this as an additional bean.
If you look at `NarayanaJtaProcessor`, that one registers the producers bean (and hence all the beans it produces), so you may want to add as well.
See https://github.com/quarkusio/quarkus/blob/master/extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java#L54